x-pack/metricbeat/module/aws/cloudwatch/metadata.go

```diff
@@ -21,15 +22,26 @@ const (
 )
 
 // addMetadata adds metadata to the given events map based on namespace
-func addMetadata(namespace string, endpoint string, regionName string, awsConfig awssdk.Config, fips_enabled bool, events map[string]mb.Event) map[string]mb.Event {
+func addMetadata(namespace string, endpoint string, regionName string, awsConfig awssdk.Config, fipsEnabled bool, events map[string]mb.Event) (map[string]mb.Event, error) {
 	switch namespace {
 	case namespaceEC2:
-		return ec2.AddMetadata(endpoint, regionName, awsConfig, fips_enabled, events)
+		events, err := ec2.AddMetadata(endpoint, regionName, awsConfig, fipsEnabled, events)
+		if err != nil {
+			return events, fmt.Errorf("error adding metadata to ec2: %w", err)
+		}
 	case namespaceRDS:
-		return rds.AddMetadata(endpoint, regionName, awsConfig, fips_enabled, events)
+		events, err := rds.AddMetadata(endpoint, regionName, awsConfig, fipsEnabled, events)
+		if err != nil {
+			return events, fmt.Errorf("error adding metadata to rds: %w", err)
+		}
 	case namespaceSQS:
-		return sqs.AddMetadata(endpoint, regionName, awsConfig, fips_enabled, events)
+		events, err := sqs.AddMetadata(endpoint, regionName, awsConfig, fipsEnabled, events)
+		if err != nil {
+			return events, fmt.Errorf("error adding metadata to sqs: %w", err)
+		}
 	default:
-		return events
+		return events, nil
 	}
+
+	return nil, fmt.Errorf("no events to add metadata to")
```

Comment:
I don't think we need this return at the end. This will make the code return warning everytime:
```
{"log.level":"warn","@timestamp":"2022-04-27T14:45:41.421-0600","log.logger":"aws.ec2","log.origin":{"file.name":"cloudwatch/cloudwatch.go","file.line":213},"message":"could not add metadata to events: %wno events to add metadata to","service.name":"metricbeat","ecs.version":"1.6.0"}
```